crates/test/mock-consensus/src/builder/init_chain.rs

```diff
@@ -0,0 +1,94 @@
+use {
+    super::*,
+    anyhow::{anyhow, bail},
+    std::time,
+    tap::TapFallible,
+    tendermint::{
+        block,
+        consensus::{
+            self,
+            params::{AbciParams, ValidatorParams, VersionParams},
+        },
+        evidence,
+        v0_37::abci::{ConsensusRequest, ConsensusResponse},
+    },
+    tower::{BoxError, Service, ServiceExt},
+    tracing::{debug, error},
+};
+
+impl Builder {
+    /// Consumes this builder, using the provided consensus service.
+    pub async fn init_chain<C>(self, mut consensus: C) -> Result<TestNode<C>, anyhow::Error>
+    where
+        C: Service<ConsensusRequest, Response = ConsensusResponse, Error = BoxError>
+            + Send
+            + Clone
+            + 'static,
+        C::Future: Send + 'static,
+        C::Error: Sized,
+    {
+        use tendermint::v0_37::abci::response;
+
+        let request = Self::init_chain_request();
+        let service = consensus
+            .ready()
+            .await
+            .tap_err(|error| error!(?error, "failed waiting for consensus service"))
+            .map_err(|_| anyhow!("failed waiting for consensus service"))?;
```

Comment:
can this use .context so we don't throw away the error info?

Reply:
then i think we also wouldn't need `tap`, since all the context would be in the error itself

Reply:
i was having difficulty with this, i'd love to figure out how to recover this too.

`anyhow::Context` seems to want a `'static` bound that isn't included in `BoxError`. if i try to call `context`, i see:

```
error[E0599]: the method `context` exists for enum `Result<&mut C, Box<dyn Error + Send + Sync>>`, but its trait bounds were not satisfied
  --> crates/test/mock-consensus/src/builder/init_chain.rs:38:14

   = note: the following trait bounds were not satisfied:
           `Box<dyn std::error::Error + Send + Sync>: anyhow::context::ext::StdError`
           which is required by `Result<&mut C, Box<dyn std::error::Error + Send + Sync>>: anyhow::Context<&mut C, Box<dyn std::error::Error + Send + Sync>>`
```